HPCC-17930 Provide couchbase plugin sample queries

rpastrana opened the pull request:
- Provides sample queries based on couchbase plugin and sample data

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana commented:
@dcamper please take a look. There's a couple of things I wasn't sure about I'll comment in-line

rpastrana commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,112 @@
+IMPORT couchbase;
+/******************************************************************************************
+   This set of example queries are based on the couchbase sample bucket 'travel-sample'
```
This is a read-only sample based on couchbase sample data, I plan on providing another set of queries based on iot data

rpastrana commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,112 @@
+    real lon;
+END;
+
+airportrec := RECORD
```
travelrec and airportrec have overlap, I'm not sure is it's possible/worthwhile to create a base and inherit

dcamper replied:
I don't think it would be worthwhile to create a base record definition. That would complicate the example unnecessarily, I think.

rpastrana commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,112 @@
+    select mybucketalias.* from `travel-sample` as mybucketalias limit 500;
+ENDEMBED;
+
+dataset(namerec) airlinenames() := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1), operation_timeout(5.5), config_total_timeout(15))
```
I'm only interested in the set of strings resulting from this query, if I declare string return, it makes it a scalar result. If I use travelrec, I'll have a lot of superfluous fields, what's the right way to do this?

dcamper replied:
This example dataset doesn't appear to have data that would be amenable to a SET OF STRING result.  We need an element in the database like:

{"address": ["123 Main St.", "Anytown, XX 98765"]}

rpastrana commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,112 @@
+    select mybucketalias.* from `travel-sample` as mybucketalias where mybucketalias.type = 'airport' and country = 'United States' limit 500;
+ENDEMBED;
+
+dataset(airportrec) airportrecordsbycountry(string m) := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1), operation_timeout(5.5), config_total_timeout(15))
```
This doesn't work, it reports that it can't bind the param, I think it has to do w/ the fact it's a string...

dcamper replied:
Binding by string works, but you have to supply the quotes.  I'll insert an inline comment below.

richardkchapman commented:
Should this be in regression suite? And any reason it can't target 6.4?

rpastrana commented:
Yes, I'll be adding to the regression suite, I haven't worked out how the couchbase server is setup.
It originally targeted 6.4, but I changed because I didn't think we'd have time to merge in... I fully expect it to be in 6.4.2 at the latest.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 26c97ec3f95c8ddecc42333fa48e992db0a87b16
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 75 | 45.345 seconds |


Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |742 |742 |0 |
|test (thor) |637 |637 |0 |
|test (roxie) |770 |770 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

dcamper commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,112 @@
+(
+    OUTPUT(airlinenames());
+    OUTPUT(usairportrecords());
+    //OUTPUT(airportrecordsbycountry('United States'));
```
Change this to make it work:

OUTPUT(airportrecordsbycountry('"United States"'));

dcamper reviewed (changes requested):
Overall, this looks great.  Some global recommendations:

* Change all "limit 500" instances to "limit 100" because the OUTPUT() calls cap the maximum rows to 100 anyway.

* In every OUTPUT() add a NAMED() flag so you can more easily see which result is for which call.  For example: OUTPUT(airlinenames(), NAMED('airlinenames'));

rpastrana committed:
HPCC-17930 Provide couchbase plugin sample queries

- Provides sample queries based on couchbase plugin and sample data
- Provides regression suite cases
- Provides regression suite keys
- Added license
- Added tags required for smokebuilds
- Trimmed and reformatted test case keys
- Altered test cases to avoid data inconsistencies
- Creates sample data on preexisting indexed data bucket
- Sets server to localhost
- Applies sleep to correct location
- Removes extra characters

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@dcamper please review
@AttilaVamos please review testing/regress/ecl/couchbase-simple.ecl and  testing/regress/ecl/key/couchbase-simple.xml and

HPCCSmoketest commented:
Automated Smoketest: :x:
Sha: 7fe5156c899bde3f8a9c5c24c634631bd34dd713
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 75 | 44.317 seconds |


Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |743 |742 |1 |
|test (thor) |638 |637 |1 |
|test (roxie) |771 |770 |1 |


[Errors:](https://gist.github.com/cb913c1dd5bd8cdbbae3137225874beb)
- test
  - hthor
    - 109\. [couchbase-simple](https://gist.github.com/HPCCSmoketest/cb913c1dd5bd8cdbbae3137225874beb#file-couchbase-simple-w20170713-182021-6-log), 
  - thor
    - 98\. [couchbase-simple](https://gist.github.com/HPCCSmoketest/cb913c1dd5bd8cdbbae3137225874beb#file-couchbase-simple-w20170713-182332-log), 
  - roxie
    - 111\. [couchbase-simple](https://gist.github.com/HPCCSmoketest/cb913c1dd5bd8cdbbae3137225874beb#file-couchbase-simple-w20170713-182942-4-log), 

HPCC Stop: OK
HPCC Uninstall: OK

AttilaVamos commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,112 @@
+   This set of example queries are based on the couchbase sample bucket 'travel-sample'
+   The sample data is easily set-up, pelase refer to the couchbase documentation for set-up
+*******************************************************************************************/
+
```
I have couple of comments:

1. Because this test is a plugin/3rd party server test you should insert two tags into this file (practically here):
//class=embedded
//class=3rdparty

This tags used by the OBT/Smoketest engines to include/exclude those test based on the configuration of the environment.

2. Actually Smoketest doesn't configured to test plugins/3rdprty servers (except memcached which is [unfortunately] integrated into ESP server), but the OBT does.

3. I need information about expectations towards/configuration of Couchbase server if they are differs from out-of-box installation.

rpastrana replied:
Done. Also sent email regarding couchbase information.

dcamper commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,312 @@
+
+/******************************************************************************************
+   This set of example queries are based on the couchbase sample bucket 'travel-sample'
+   The sample data is easily set-up, pelase refer to the couchbase documentation for set-up
```
Typo: 'pelase' should be 'please'

dcamper commented on initfiles/examples/embed/couchbase-simple.ecl:
```diff
@@ -0,0 +1,312 @@
+ENDEMBED;
+
+dataset(routerec) fullroute()  := EMBED(couchbase : server(server), bucket(thebucket), detailed_errcodes(1), operation_timeout(5.5), config_total_timeout(15))
+    SELECT mybucketalias.* FROM `travel-sample` as mybucketalias WHERE type = 'route' and sourceairport IS NOT NULL LIMIT 11
```
LIMIT 11?  Should that be LIMIT 1?

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 10bbf45ebd2134c73ae1e74052423d571c941381
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 96.2 seconds |


Install hpccsystems-platform-community_6.4.0-rc5.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :x:
Sha: 45f15e20e1bb21b17edaf3d4b8d7aeea027aaece
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 93.451 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |759 |1 |


[Errors:](https://gist.github.com/df1cb0c4155d027ce14fa1636599b1c0)
- test
  - roxie
    - 524\. [schedule2](https://gist.github.com/HPCCSmoketest/df1cb0c4155d027ce14fa1636599b1c0#file-schedule2-w20170719-151654-3-log), 

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :x:
Sha: 45f15e20e1bb21b17edaf3d4b8d7aeea027aaece
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.887 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |759 |1 |


[Errors:](https://gist.github.com/a0c29ede5bd0953a10104026e3b8b5ce)
- test
  - roxie
    - 524\. [schedule2](https://gist.github.com/HPCCSmoketest/a0c29ede5bd0953a10104026e3b8b5ce#file-schedule2-w20170719-162454-3-log), 

HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 8102e7fe686e7fd03ce3acfae095703999085f63
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.862 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 43abef56302c2efde2ad3818a30f2e6b0c2e4639
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.673 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

kunalaswani commented:
![image](https://user-images.githubusercontent.com/6652898/28419503-1c381fb8-6d2d-11e7-81d4-f5b2692053d4.png)

couchbase regression test succeeds on runtime

rpastrana commented:
@dcamper please take a quick look again.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 618c7764ae6a395e5ab4ab91be5ae34157c2a556
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 94.926 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

dcamper commented:
Looks good.

rpastrana commented:
@richardkchapman please merge

rpastrana commented:
@dcamper please review 2nd set of sample queries

dcamper commented on initfiles/examples/embed/couchbase-simple2.ecl:
```diff
@@ -0,0 +1,140 @@
+      create the iot bucket on the target couchbase server
+************************************************************************************************/
+
+server := '10.176.152.227';
```
Should this IP address be 127.0.0.1 instead?

AttilaVamos replied:
Yes, it should. This test should run in any machine inside and outside.  The better solution would be this:
```
server := '.' : stored('ServerIP');
```
or
```
server := '127.0.0.1' : stored('ServerIP');
```
In this case the default is the localhost, but if user wants to execute this test with a Couchbase server on a different IP address, then they can do it with -X parameter in this command:
```
./ecl-test query -X ServerIP=<couchbase_server_ipaddress> couchbase-simple2.ecl
```
with the IP above:
```
./ecl-test query -X ServerIP=10.176.152.227 couchbase-simple2.ecl
```

rpastrana replied:
Good point. This sample script isn't meant to run as part of the regress suite.

dcamper commented on initfiles/examples/embed/couchbase-simple2.ecl:
```diff
@@ -0,0 +1,140 @@
+  OUTPUT(insertiotdoc(adocid + random() + '"', adoc2));
+  OUTPUT(upsertiotdoc(adocid + random() + '"', adoc1));
+  OUTPUT(getlatesttimestamp());
+  Std.System.Debug.Sleep(200);
```
Might want to move this Sleep() call up one line so it sits between the updates and the reads.

dcamper commented on initfiles/examples/embed/couchbase-simple2.ecl:
```diff
@@ -0,0 +1,140 @@
+  OUTPUT(devicebylocation(ROW({2.247,33.629,1.323,'"A"'},locationDatarec)));
+  OUTPUT(deviceidsbyzone('"A"'));
+  OUTPUT('Done');
+););
```
The final ');' seems to be spurious.  Plus I think you should put a final newline at the end.

dcamper reviewed (changes requested):
Overall, it looks good. A few minor changes.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 091825d3639dc227555b9dfd5ae8944dfe9f5e22
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 91.762 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

AttilaVamos commented:
@rpastrana I have a comment for the server address.

rpastrana commented:
@dcamper one last review.

dcamper reviewed (approved):
Looks good to go.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 86d4a0c1bec76606c6564f1d8075138f7ed7e493
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 92.451 seconds |


Install hpccsystems-platform-community_6.4.0-rc6.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK

rpastrana commented:
@richardkchapman commits squashed as requested

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
Sha: 5e7502d1ea7249ddca3e94bd241b5f5b1ff32a27
Build: success
Build: success
ECL Watch:  Rebuilding Site

| errors | warnings | build time | 
|---|---|---|
| 0 | 65 | 94.05 seconds |


Install hpccsystems-platform-community_6.4.1-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |95 |95 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |733 |733 |0 |
|test (thor) |627 |627 |0 |
|test (roxie) |760 |760 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
